codegen/templates/http_client.tmpl

```diff
@@ -129,16 +133,24 @@ func (c *{{$clientName}}) {{title .Name}}(
 	switch res.StatusCode {
 		case {{.OKStatusCode.Code}}:
 			var responseBody {{unref .ResponseType}}
+			{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(&responseBody)
+			{{else -}}
+			rawBody, err := res.ReadAll()
 			if err != nil {
-				return nil, respHeaders, err
+				return _res, respHeaders, err
+			}
+			err = json.Unmarshal(rawBody, &responseBody)
```

Comment:
`ReadAndUnmarshalBody ` expects a `json.Unmarshaler` interface, I don't think simply adding `&` will do it.